src/guide/plugins.md

```diff
@@ -12,49 +12,77 @@ Plugins usually add global-level functionality to Vue. There is no strictly defi
 
 5. A library that provides an API of its own, while at the same time injecting some combination of the above (e.g. [vue-router](https://github.com/vuejs/vue-router)).
 
-## Using a Plugin
+## Writing a Plugin
+
+In order to better understand how to create your own Vue.js plugins, we will create a very simplified version of a plugin that displays `i18n` ready strings. 
 
-After a Vue app has been initialized with `createApp()`, you can add a plugin to your application by calling the `use()` method:
+A Vue.js plugin is either an `object` that should expose an `install()` method, or a `function`. 
+
+Whenever this plugin is added to an application, the `install` method will be called if it is an object. If it is a `function`, the function itself will be called. In both cases, it will receive two parameters - the `app` object resulting from Vue's `createApp`, and the options passed in by the user.
+
+Let's begin by setting up the plugin object.
 
 ```js
```

Comment:
Where would this hypothetical file live?

```suggestion
```js
// /User/i18nPlugin/src/index.js
```

Reply:
The way the documentation was previously written implied that the code was being written directly on top of what looks like a `main.js` file. Do we want to incorporate some instructions on how to make it into its own file and import it? I can add that

Reply:
@marina-mosti Definitely. I think it would be super helpful for those who have no context as to where to start.